### ISSUE TITLE
📣 Upcoming changes to the lists feature

### ISSUE THREAD

**Issue by Witoso** [ckeditor/ckeditor5#14767]
We wanted to inform you about some essential changes coming to the list feature in CKEditor 5.

## TL;DR

*   Over the years, we developed two independent list feature implementations.
    *   [`List`](https://ckeditor.com/docs/ckeditor5/latest/features/lists/lists.html) (later called List v1) 
    *   [`DocumentList`](https://ckeditor.com/docs/ckeditor5/latest/features/lists/document-lists.html#demo) (later called List v2).
*   List v2 is more future-proof and feaure-rich.
*   We need to sunset List v1. But there are small gaps in feature-parity, which we are fixing now.
*   This will be a breaking change to the majority of you, but the migration path will be trivial.
*   Stay tuned. No action needed yet.

## Context

We currently maintain two list features: [`List`](https://ckeditor.com/docs/ckeditor5/latest/features/lists/lists.html) (later called List v1) and [`DocumentList`](https://ckeditor.com/docs/ckeditor5/latest/features/lists/document-lists.html#demo) (later called List v2).

The List v1 feature was implemented in the early days of CKEditor 5. It supports “plain lists” – lists where `<li>` cannot contain block content (paragraphs, headings, tables, block images). It supports todo lists. But it does not support extending list markup via the [General HTML Support](https://ckeditor.com/docs/ckeditor5/latest/features/html/general-html-support.html) (GHS) feature.

The List v2 feature was implemented in 2022 to add support for block content in list items. It supports extending list markup via GHS. It does not support todo lists.

All development effort goes into List v2. The time has come to sunset List v1. Keeping two features that are almost the same is a waste of time and causes confusion for integrators. At the same time, we want to share new list features with a larger user base!

## Plan overview

*   \[DONE\] Stage 1. **Ensure feature completeness of List v2**
    *   Scope:
        *   Add support for todo lists.
        *   Add an option to configure List v2 to behave like List v1 (allow only inline content). It’s an edge case, but we aim at making sure that the migration path is trivial.
        *   Issue: #14632.
*   Stage 2. **Replacement (beginning of 2024, to be confirmed)**
    *   Scope:
        *   `List` plugin with be replaced with `DocumentList` implementation.
            *   Effect: after the update, without any action, you will start using List v2.
        *   `LegacyList` plugin will be released as a new separate plugin if someone wants to use the old lists implementation and have more time to migrate.
        *   `DocumentList` will still work, but will issue a deprecation warning.
*   Stage 3. **Sunset (to be scheduled)** 
    *   Scope:
        *   Removal of the `DocumentList` and `LegacyList` plugins.

## Stay tuned

We’ll use this ticket to notify you about the status of the project.

Keep in mind that you can use the nightly releases (see `0.0.0-*` versions on NPM) to test any changes before the final releases hit the registry.

**Comment by Witoso** [comment]
Small update for those watching, todo list and simple lists config are in progress. The replacement is to be started. Both are planned to be a part of one release.

**Comment by Witoso** [comment]
Update: replacement of `List` to `DocumentList` will happen in the January release. Stay tuned!

**Comment by Witoso** [comment]
Closing stale ticket that I missed :grimacing: , released in #15037.

**Comment by Witoso** [comment]
Stage 3. **Sunset**

*   Scope:
    *   Removal of the `DocumentList` and `LegacyList` plugins.

Will most likely happen next year, stay tuned, we will inform about this.
